docs/src/design/proposals/Registration.md

```diff
@@ -0,0 +1,518 @@
+# Introduction
+
+This document proposes a Registration API which allows for components
+of Open MCT to communicate with one another using reusable, internally
+standardized patterns.
+
+# Developer Use Cases
+
+1. Extending and maintaining Open MCT itself.
+2. Adapting and customizing Open MCT for use in specific missions.
+3. Developing features for use with Open MCT across multiple different
+   missions.
+
+# Scope
+
+As demonstrated by the existing APIs, writing plugins is sufficient to
+satisfy the three use cases above in the majority of cases. The only feature
+which is known to be unsatisfiable by plugins is plugin support itself.
+
+As such, prefer to keep "plugin-external" components small and simple, to
+keep the majority of development in plugins (or plugin-like components.)
+
+The "registration API" described in this document is limited to that scope:
+It describes classes and patterns that can allow plugins to interact,
+while making minimal assumptions about what specific functionality is to
+be implemented in these plugins.
+
+By analogy to current API, this set of functionality will effectively
+replace the [Framework Layer](http://nasa.github.io/openmctweb/guide/#framework)
+of Open MCT.
+
+# Problems to Address
+
+1. Dependencies between plugins are implicit; a plugin may fail if its
+   required dependencies are not included, without any clear indication
+   of why this failure has occurred. Significant familiarity with
+   Open MCT is typically required to debug in these circumstances.
+2. Extension points are often implicit; no specific plugin is
+   identifiably responsible for defining any specific extension category
+   or composite service. These are instead paired by string matching.
+   This makes it difficult to follow how the application is initialized
+   and how objects are passed around at run-time, and creates an
+   additional documentation burden, as these named extension points do not
+   fit into more standard API documentation.
+3. Reuse of components between plugins is limited. Exposing base classes
+   from one plugin and reusing them from another is overly-difficult.
+
+# Principles
+
+1. The Registration API is exposed as a set of classes in one or more
+   namespaces. This supports reuse and extension using standard,
+   well-known object-oriented patterns. A composition-oriented style
+   is still supported and encouraged, but not enforced.
+2. Extension points should be expressed as objects with known interfaces.
+   (More specifically, they should be _expressible_ in this fashion;
+   it is out of scope for the Registration API to expose any specific
```

Comment:
Core capabilities: things that are part of the core platform and have no business being removed/turned off.  These are what make the platform what it is.
Plugins: things which extend the platform in a very specific manner.

`./platform/framework/bundle` - ***core capability***
`./platform/core/bundle` - ***core capability***
`./platform/representation/bundle` - ***core capability***
`./platform/commonUI/about/bundle` - ***core capability***
`./platform/commonUI/browse/bundle` - ***core capability***
`./platform/commonUI/edit/bundle` - ***core capability***
`./platform/commonUI/dialog/bundle` - ***core capability***
`./platform/commonUI/formats/bundle` - ***core capability***
`./platform/commonUI/general/bundle` - ***core capability***
`./platform/commonUI/inspect/bundle` - ***core capability***
`./platform/commonUI/mobile/bundle` - ***core capability***
`./platform/commonUI/themes/espresso/bundle` - ***platform configuration option***
`./platform/commonUI/notification/bundle` - ***core capability***
`./platform/containment/bundle` - ***core capability***
`./platform/execution/bundle` - ***core capability***
`./platform/exporters/bundle` - ***core capability***
`./platform/telemetry/bundle` - ***core capability***
`./platform/features/clock/bundle` - ***plugin***
`./platform/features/imagery/bundle` - ***plugin***
`./platform/features/layout/bundle` - ***plugin***
`./platform/features/pages/bundle` - ***plugin***
`./platform/features/plot/bundle` - ***plugin***
`./platform/features/timeline/bundle` - ***plugin*** (unfortunate dependency on plot)
`./platform/features/table/bundle` - ***plugin***
`./platform/forms/bundle` - ***core capability***
`./platform/identity/bundle` - ***core capability***
`./platform/persistence/aggregator/bundle` - ***core capability***
`./platform/persistence/local/bundle` - ***plugin***
`./platform/persistence/queue/bundle` - ***plugin***
`./platform/policy/bundle` - ***core capability***
`./platform/entanglement/bundle` - ***core capability***
`./platform/search/bundle` - ***core capability*** (generic search should be split off as a plugin)
`./platform/status/bundle` - ***core capability***
`./platform/commonUI/regions/bundle` - ***core capability***